### PR TITLE
Animate page editor sidebar

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

Properly animates the page editor's left-hand sidebar to match the right-hand one, both in behavior and in component style. This should also make it a bit easier to add [resizable panes](https://github.com/pixiebrix/pixiebrix-extension/issues/4867) as also [suggested by Britt](https://pixiebrix.slack.com/archives/C042CA15AKS/p1706643177213929)

This is also part of the effort to create smaller components, as it splits the sidebar UI logic from the extension list logic.

This code is pretty old, I wrote it while testing out React's [hot-module reloading](https://github.com/pixiebrix/pixiebrix-extension/pull/7598)

## Before

https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/ac848ea9-9d7f-44b7-8a9e-88f3679e2537

## After

https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/b2e5431c-bb54-428c-9bf3-1ce74f9a8252

## Checklist

- [ ] Add tests and/or storybook stories
- [x] Designate a primary reviewer: @grahamlangford 

## Related

- https://github.com/pixiebrix/pixiebrix-extension/pull/7865
- https://github.com/pixiebrix/pixiebrix-extension/issues/4867

### CODE REVIEW

**Review by BrandonPxBx — COMMENTED** [review]
Animations look pretty great! It's much easier to follow what's happening in the panes. One thing that jumped out for me was how the larger data panel animation feels much faster than the mod panel. 

I bet the animation timing is the same, but since the data panel is larger it feels faster. If we decrease the total animation time for the mod panel, to about 60% of what it is right now, it would make them feel more uniform.

**Comment by fregante** [comment]
> feels much faster than the mod panel

Matched:

https://github.com/pixiebrix/pixiebrix-extension/blob/329248f85a8f11203ee3758debd5ea8f53970f0e/src/pageEditor/sidebar/Sidebar.module.scss#L14-L17


https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/4b958b90-d743-4256-b885-09fc3e2a82f4

Speaking of the data panel, a while ago I had attempted to make the it a bit tighter, there's a lot of wasted space due to the toggle. Also I patched up the broken "tab" look

_Not part of this PR_


Before:

Screenshot 7

After:

Screenshot 6

File: src/pageEditor/sidebar/Sidebar.module.scss [hunk 1]
```diff
@@ -1,127 +1,64 @@
...
+  box-shadow: 1px 2px 8px #00000030;
+  z-index: 999; // Always on top of the content, so that the shadow is applied correctly
+
+  :global(.collapsing) {
+    // Match the speed of the animation with the righthand sidebar
+    // https://github.com/pixiebrix/pixiebrix-extension/pull/8125#pullrequestreview-1971309628
+    // Math:
+    // Righthand: 360px (collapsable width) -> 0.35s (Bootstrap default)
+    // Lefthand: 240px (collapsable width) -> 0.23s
+    transition-duration: 0.23s;
```
Comment (grahamlangford) [inline src/pageEditor/sidebar/Sidebar.module.scss:17]: Can't we just pass in the `timeout` to the Bootstrap Collapse element

Comment (fregante) [reply src/pageEditor/sidebar/Sidebar.module.scss:17]: Unfortunately not, the timeout is just a timeout (i.e. consider the transition completed even without `transitionend` event), it does not control the animation.

Comment (grahamlangford) [reply src/pageEditor/sidebar/Sidebar.module.scss:17]: Well that's annoying.

**Review by grahamlangford — APPROVED** [review]

**Review by BrandonPxBx — APPROVED** [review]

**Pull request merged**